packages/next-swc/crates/core/src/emotion/mod.rs

```diff
@@ -0,0 +1,842 @@
+                                                        spread: None,
+                                                        expr: Box::new(Expr::Lit(Lit::Str(
+                                                            self.create_label(true).into(),
+                                                        ))),
+                                                    });
+                                                }
+                                                if let Some(cm) =
+                                                    self.create_sourcemap(tagged_tpl.span.lo())
+                                                {
+                                                    args.push(ExprOrSpread {
+                                                        spread: None,
+                                                        expr: Box::new(Expr::Lit(Lit::Str(
+                                                            cm.into(),
+                                                        ))),
+                                                    });
+                                                }
+                                                args
+                                            },
+                                            type_args: None,
+                                        });
+                                    }
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        expr.fold_children_with(self)
+    }
+
+    fn fold_jsx_element(&mut self, mut expr: JSXElement) -> JSXElement {
+        match &mut expr.opening.name {
+            JSXElementName::Ident(i) => {
+                if let Some(PackageMeta::Named(ExprKind::GlobalJSX)) =
+                    self.import_packages.get(i.as_ref())
+                {
+                    self.rewrite_styles_attr(&mut expr.opening.attrs, i.span.lo());
+                }
+            }
+            JSXElementName::JSXMemberExpr(member_exp) => {
+                if let JSXObject::Ident(i) = &member_exp.obj {
+                    if let Some(PackageMeta::Namespace(EmotionModuleConfig {
+                        exported_names,
+                        ..
+                    })) = self.import_packages.get(i.as_ref())
+                    {
+                        if exported_names.iter().any(|item| {
+                            matches!(item.kind, ExprKind::GlobalJSX)
+                                && item.name == member_exp.prop.as_ref()
+                        }) {
+                            self.rewrite_styles_attr(&mut expr.opening.attrs, i.span.lo());
+                        }
+                    }
+                }
+            }
+            _ => {}
+        };
+        let dest_expr = expr.fold_children_with(self);
+        self.in_jsx_element = false;
```

Comment:
You can store the original field value and restore it, or use usize.

Reply:
> This can be problematic for nested jsx expressions.

It's ok in this scenario. The `is_jsx_element` flag is for determine whether to process `css()` or `css`\`\` expressions. `css` in `<div>{}</div>` is need to be processed.

Reply:
Then it's fine, thanks for clarification!